src/components/DashboardHeader/DashboardHeader.tsx

```diff
@@ -26,7 +22,7 @@ export class DashboardHeader extends React.Component<IDashboardHeaderProps, any>
                         <Icon className={'add-farm'} iconName={'icon-add'} onClick={this.props.onAddFarmClick} title={'Add'} />
                     }
                 </span>
-                <Search onSearch={this.props.onSearch} onChange={this.props.onChanged} value={this.props.filter} />
+                <Search onSearch={this.props.onSearch} onChange={_.debounce(this.props.onChanged, 250)} />

```

Comment:
i believe that the debounce should occur within the Search component, so if it is used elsewhere we do not have the same problem